joomla/joomla-cms
# [4] Generic Blank State for Categories of Extensions & Tags

### Summary of Changes

Introducing a generic blank state that is reusable for any extension.

Tags and categories are unique with blank states as their db tables are NOT EMPTY when a blank state is true. They require the tables to be additionally filtered by a column. This PR improves the initial work on blank states to provide that additional filtering generically so it can be reused. 

### Testing Instructions

Delete your categories for "Article Categories" while maintaining some "News feed categories" (remembering to empty trash)

Navigate to Content -> Categories  = See new blank state
Navigate to Components -> News Feed Categories  = See your existing news feed categories
Trash all your Newsfeed categories, empty trash = See new blank state

### Actual result BEFORE applying this Pull Request

No blank states, only "No matching Results" screen historically. 

### Expected result AFTER applying this Pull Request

A generic categories screen, reusing existing language strings from components. 

<img width="651" alt="Screenshot 2021-04-24 at 19 28 31" src="https://user-images.githubusercontent.com/400092/115969112-438a8a00-a533-11eb-9434-4a73c0523f37.png">

<img width="586" alt="Screenshot 2021-04-24 at 19 28 41" src="https://user-images.githubusercontent.com/400092/115969122-49806b00-a533-11eb-99c6-701233d2ef69.png">

<img width="822" alt="Screenshot 2021-04-24 at 20 02 40" src="https://user-images.githubusercontent.com/400092/115970073-1096c500-a538-11eb-83bc-c2b09c564d8c.png">


### Documentation Changes Required
